cmd/kubernetesDeploy.go

```diff
@@ -0,0 +1,232 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+func kubernetesDeploy(myKubernetesDeployOptions kubernetesDeployOptions) error {
+	c := command.Command{}
+	// reroute stderr output to logging framework, stdout will be used for command interactions
+	c.Stderr(log.Entry().Writer())
+	runKubernetesDeploy(myKubernetesDeployOptions, &c, log.Entry().Writer())
+	return nil
+}
+
+func runKubernetesDeploy(myKubernetesDeployOptions kubernetesDeployOptions, command envExecRunner, stdout io.Writer) {
+	if myKubernetesDeployOptions.DeployTool == "helm" {
+		runHelmDeploy(myKubernetesDeployOptions, command, stdout)
+	} else {
+		runKubectlDeploy(myKubernetesDeployOptions, command)
+	}
+}
+
+func runHelmDeploy(myKubernetesDeployOptions kubernetesDeployOptions, command envExecRunner, stdout io.Writer) {
+	_, containerRegistry, err := splitRegistryURL(myKubernetesDeployOptions.ContainerRegistryURL)
+	if err != nil {
+		log.Entry().WithError(err).Fatalf("Container registry url '%v' incorrect", myKubernetesDeployOptions.ContainerRegistryURL)
+	}
+	containerImageName, containerImageTag, err := splitFullImageName(myKubernetesDeployOptions.Image)
+	if err != nil {
+		log.Entry().WithError(err).Fatalf("Container image '%v' incorrect", myKubernetesDeployOptions.Image)
+	}
+	helmLogFields := map[string]interface{}{}
+	helmLogFields["Chart Path"] = myKubernetesDeployOptions.ChartPath
+	helmLogFields["Namespace"] = myKubernetesDeployOptions.Namespace
+	helmLogFields["Deployment Name"] = myKubernetesDeployOptions.DeploymentName
+	helmLogFields["Context"] = myKubernetesDeployOptions.KubeContext
+	helmLogFields["Kubeconfig"] = myKubernetesDeployOptions.KubeConfig
+	log.Entry().WithFields(helmLogFields).Debug("Calling Helm")
+
+	helmEnv := []string{fmt.Sprintf("KUBECONFIG=%v", myKubernetesDeployOptions.KubeConfig)}
+	log.Entry().Debugf("TILLER_NAMESPACE=%v", myKubernetesDeployOptions.TillerNamespace)
```

Comment:
Obsolete in my eyes as `helmEnv` is reported some lines below.

Reply:
good catch!